### PR TITLE
[CHI-297] CE - Alert

### PR DESCRIPTION
**Pull request by srinath-koneru**
Progress:

- Custom element
- Documentation
- Cypress end-to-end tests
- Backstop tests

This PR depends on #351

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/178/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/179/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/180/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/181/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/182/ for more details.

**Comment by srinath-koneru** [comment]
@dani-cl-madrid @SergioQCL this PR is ready for review. TIA

File: src/website/views/custom-elements/alert.pug [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  | To render an alert, use the tag <code>&lt;chi-alert&gt;</code>.
+  | A value (e.g. <code>warning</code>) must be set to attribute <code>color</code> so as to provide proper styling.
+
+h4 Base
+.example.-mb--3
+  .-p--1
+    each type in ['success', 'danger', 'warning', 'info', 'muted']
+      chi-alert.-m--2(color=type)= `This is a ${type} alert`
+  :code(lang="html")
+    <chi-alert color="success" >This is a success alert</chi-alert>
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]: Please, remove extra space between last property and _>_, and fix in further occurrences.

File: src/website/views/custom-elements/alert.pug [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    <chi-alert color="danger" >This is a danger alert</chi-alert>
+    <chi-alert color="warning" >This is a warning alert</chi-alert>
+    <chi-alert color="info" >This is a info alert</chi-alert>
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]:
```suggestion
    <chi-alert color="info" >This is an info alert</chi-alert>
```

File: src/website/views/custom-elements/alert.pug [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      <chi-icon icon="circle-warning" color="danger" size="sm--2" ></chi-icon>
+      This is a danger alert
+    </chi-alert>
+    <chi-alert color="warning" >
+      <chi-icon icon="warning" color="warning" size="sm--2" ></chi-icon>
+      This is a warning alert
+    </chi-alert>
+    <chi-alert color="info" >
+      <chi-icon icon="circle-info" color="primary" size="sm--2" ></chi-icon>
+      This is a info alert
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]:
```suggestion
      This is an info alert
```

File: src/website/views/custom-elements/alert.pug [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    </chi-alert>
+    <chi-alert color="muted" >
+      <chi-icon icon="circle-info" color="muted" size="sm--2" ></chi-icon>
+      This is a muted alert
+    </chi-alert>
+
+h5 Icon Center
+p.-text Use the attribute <code>center</code> to center the icons.
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]:
```suggestion
p.-text Use the <code>center</code> attribute  to center the icons.
```

File: src/website/views/custom-elements/alert.pug [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      chi-icon(icon='circle-info', color='primary', size='sm--2')
+      = 'Info alert with centered icon'
+  :code(lang="html")
+    <chi-alert color="info" center >
+      <chi-icon icon="circle-info" color="primary" size="sm--2" ></chi-icon>
+      Info alert with centered icon
+    </chi-alert>
+
+h4 Dismissible
+p.-text Use the attribute <code>dismissible</code> to allow the user to dismiss the alert.
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]:
```suggestion
p.-text Use the <code>dismissible</code> attribute to allow the user to dismiss the alert.
```

File: src/website/views/custom-elements/alert.pug [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+      <div>
+        <div class="a-h5">warning</div>
+        This is a warning alert
+      </div>
+    </chi-alert>
+    <chi-alert type="toast" color="info">
+      <chi-icon size="sm--2" icon="circle-info" color="primary" ></chi-icon>
+      <div>
+        <div class="a-h5">info</div>
+        This is a info alert
```
Comment (SergioQCL) [inline src/website/views/custom-elements/alert.pug]:
```suggestion
        This is an info alert
```

File: src/custom-elements/src/components/alert/alert.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+export class Alert {
+  /**
+   *  to set alert type { bubble, banner, toast }.
+   */
+  @Prop({ reflectToAttr: true }) type = 'bubble';
+
+  /**
+   *  to set alert contextual class { success, danger, warning, info, muted }.
+   */
+  @Prop({ reflectToAttr: true }) color!: string;
```
[the remark this answers is not in the document]
Comment (srinath-koneru) [reply src/custom-elements/src/components/alert/alert.tsx]: @SergioQCL It states that this prop is required by the element.

File: src/custom-elements/src/components/alert/alert.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+  /**
+   *  to set size of an alert, only value used is lg.
+   */
+  @Prop({ reflectToAttr: true }) size: string;
```
[the remark this answers is not in the document]
Comment (srinath-koneru) [reply src/custom-elements/src/components/alert/alert.tsx]: @SergioQCL It is the only available size but certainly not the default one.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/186/ for more details.

File: src/custom-elements/src/components/alert/alert.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (srinath-koneru) [reply src/custom-elements/src/components/alert/alert.tsx]: @SergioQCL I was thinking on the same lines but I was a little skeptical on going against the standards. ```size``` with one value was superior than using ```large``` or ```size``` attribute as boolean for me. Thoughts @dani-cl-madrid @jllr.

[the remark this answers is not in the document]
Comment (SergioQCL) [reply src/custom-elements/src/components/alert/alert.tsx]: Ok. Let's keep size

File: src/custom-elements/src/components/alert/alert.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        role="alert"
+      >
+        {(this.dismissible || this.type === 'toast' || this.size === 'lg') ?
+          <div class="m-alert__content">
+            <slot></slot>
+          </div>
+          :
+          <slot></slot>
+        }
+        {(this.dismissible || this.type === 'toast') && <chi-button type="close" onChiClick={this._dismissAlert.bind(this)} size={'sm'} />}
```
[the remark this answers is not in the document]
Comment (srinath-koneru) [reply src/custom-elements/src/components/alert/alert.tsx]: @dani-cl-madrid @SergioQCL I'm inclined towards ```mutable```.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/211/ for more details.

File: src/custom-elements/src/components/alert/alert.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (SergioQCL) [reply src/custom-elements/src/components/alert/alert.tsx]: Yes, mutable makes sense

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/217/ for more details.

**Comment by srinath-koneru** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/218/ for more details.

**Comment by srinath-koneru** [comment]
@dani-cl-madrid made the requested changes.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/220/ for more details.

**Pull request merged**